### PR TITLE
Gulp babili

### PR DESCRIPTION
**Pull request by boopathi**
TODO:

+ [ ] comments option doesn't work (@hzoo I don't know why shouldPrintComment is not called at all).
+ [x] README

### CODE REVIEW

File: packages/gulp-babili/__tests__/__snapshots__/gulp-babili-test.js.snap [hunk 1]
```diff
@@ -0,0 +1,5 @@
+exports[`gulp-babili comments should remove all comments when false 1`] = `"foo(),bar(),baz();"`;
```
Comment (boopathi) [inline packages/gulp-babili/__tests__/__snapshots__/gulp-babili-test.js.snap]: this output is wrong

File: packages/gulp-babili/src/index.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+const applySourceMap = require("vinyl-sourcemaps-apply");
+const babiliPreset = require("babel-preset-babili");
+
+const {PluginError} = gutil;
+
+const NAME = "gulp-babili";
+
+module.exports = gulpBabili;
+
+function gulpBabili(_opts = {}) {
```
Comment (boopathi) [inline packages/gulp-babili/src/index.js]: will change this and take 2 arguments instead for passing in custom babel, babili and comments

File: packages/gulp-babili/src/index.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      /* preset */
+      presets: [[babili, opts]],
+
+      /* sourcemaps */
+      sourceMaps: true,
+      inputSourceMap,
+
+      /* remove comments */
+      shouldPrintComment(contents) {
+        return shouldPrintComment(contents, commentsRegex);
```
Comment (boopathi) [inline packages/gulp-babili/src/index.js]: this is never called.

**Comment by kangax** [comment]
fuse-box that was just released uses gulp-uglify as well https://github.com/fuse-box/fuse-box/blob/461eeeeef5752e069b53de009554af7be145a499/gulpfile.js#L31

**Comment by kangax** [comment]
@boopathi @hzoo Guys, any chance to move this forward? Wish I could help but not familiar with Gulp at all.

File: packages/gulp-babili/__tests__/gulp-babili-test.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    let file;
+
+    beforeEach(() => {
+      file = new gutil.File({
+        path: "comments.js",
+        contents: new Buffer(source)
+      });
+    });
+
+    xit("should remove comments by default except license and preserve", () => {
```
Comment (boopathi) [inline packages/gulp-babili/__tests__/gulp-babili-test.js]: this test is disabled as it removes all comments by default.

**Pull request merged**